# fixed upstream but not yet released
# https://github.com/dgasmith/opt_einsum/commit/7c8f193f90b6771a6b3065bb5cf6ec2747af8209
From 7c8f193f90b6771a6b3065bb5cf6ec2747af8209 Mon Sep 17 00:00:00 2001
From: Hongxu Jia <hongxujia83@gmail.com>
Date: Sat, 19 Nov 2022 07:40:16 +0800
Subject: [PATCH] Use ConfigParser instead of SafeConfigParser (#208)

The SafeConfigParser class will be renamed to ConfigParser in Python
3.12 [1]. This alias will be removed in future versions.So we can use
ConfigParser directly instead.

[1] https://github.com/python/cpython/issues/89336

Signed-off-by: Hongxu Jia <hongxu.jia@windriver.com>

Signed-off-by: Hongxu Jia <hongxu.jia@windriver.com>
---
 versioneer.py | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/versioneer.py b/versioneer.py
index d3db643..6d732af 100644
--- a/versioneer.py
+++ b/versioneer.py
@@ -339,9 +339,9 @@ def get_config_from_root(root):
     # configparser.NoOptionError (if it lacks "VCS="). See the docstring at
     # the top of versioneer.py for instructions on writing your setup.cfg .
     setup_cfg = os.path.join(root, "setup.cfg")
-    parser = configparser.SafeConfigParser()
+    parser = configparser.ConfigParser()
     with open(setup_cfg, "r") as f:
-        parser.readfp(f)
+        parser.read_file(f)
     VCS = parser.get("versioneer", "VCS")  # mandatory
 
     def get(parser, name):
